.circleci/config.yml

```diff
@@ -853,6 +863,72 @@ jobs:
+      #     name: Install and configure the SRP observer tool
+      #     command: |
+      #       wget https://s3.amazonaws.com/srp-cli/linux-observer-latest.tar.gz
+      #       tar zxf linux-observer-latest.tar.gz
+      - run:
+          name: Generate the "source provenance" file
+          command: |
+            # This UID should have been already set in the internal tool.
+            PROV_OBJ_UID="uid.obj.build.circleci(instance='circleci.com',type='<< pipeline.project.type >>',namespace='${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}',branch='${CIRCLE_BRANCH}',commit_id='${CIRCLE_SHA1}',build_id='<< pipeline.number >>/${CIRCLE_WORKFLOW_ID}/${CIRCLE_BUILD_NUM}')"
+            echo "PROV_OBJ_UID=${PROV_OBJ_UID}"
+            # Using CIRCLE_TAG as the version if defined (i.e., when releasing a version); use the commit SHA otherwise.
+            VERSION=$([ -z "${CIRCLE_TAG}" ] && echo "${CIRCLE_SHA1}" || echo "${CIRCLE_TAG}")
+            srp provenance source --verbose --scm-type git --name "kubeapps" --path ./ --saveto ./source-provenance.json --comp-uid "${PROV_OBJ_UID}" --build-number "${CIRCLE_WORKFLOW_ID}" --version "${VERSION}" --all-ephemeral true --build-type release
+            cat ./source-provenance.json
+      - store_artifacts:
+          path: source-provenance.json
+      # TODO(agamez): we will need to create the "network provenance" file soon.
+      # - run:
+      #     name: Generate the "network provenance" file
+      #     command: |
+      #       JAVA_HOME="/dev/null"
+      #       mkdir provenance-go
+      #       ./bin/observer_agent -t -o ./provenance-go -- go mod download -x
+      #       cat provenance-go/provenance.json
+      #       mkdir provenance-yarn
+      #       ./bin/observer_agent -t -o ./provenance-yarn -- yarn install --cwd=dashboard --frozen-lockfile
+      #       cat provenance-yarn/provenance.json
+      #       jq -s 'flatten | group_by(.artifact_repositories) | map(reduce .[] as $x ({}; . * $x))'  provenance-go/provenance.json  provenance-yarn/provenance.json > network-provenance.json
+      #       cat network-provenance.json
+      # - store_artifacts:
+      #     path: network-provenance.json
+      # - run:
+      #     name: Merge the provenance files
+      #     command: |
+      #       srp provenance merge --verbose --source ./source-provenance.json --network ./network-provenance.json --saveto ./merged-provenance.json
+      # - store_artifacts:
+      #     path: merged-provenance.json
+      - run:
+          name: Validate and submit the provenance files to the SRP Metadata service
+          command: |
+            srp uid validate "uid.mtd.provenance_2_5.fragment(obj_uid=uid.obj.build.circleci(instance='circleci.com',type='<< pipeline.project.type >>',namespace='${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}',branch='${CIRCLE_BRANCH}',commit_id='${CIRCLE_SHA1}',build_id='<< pipeline.number >>/${CIRCLE_WORKFLOW_ID}/${CIRCLE_BUILD_NUM}'),revision='')"
```

Comment:
If there is an upstream issue or the format for CircleCI happens to change, this will throw an error.